### PR TITLE
Made the encodings of .razor and .cshtml files in the repo consistent

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Fixes https://github.com/dotnet/aspnetcore/issues/22753

- 99% of the changes in this PR is adding UTF8BOM to .cshtml and .razor files in the repo.
- See the second commit for the test fixes

### CODE REVIEW

File: src/ProjectTemplates/test/ByteOrderMarkTest.cs [hunk 1]
```diff
@@ -21,79 +21,78 @@ public ByteOrderMarkTest(ITestOutputHelper output)
             _output = output;
         }
 
-        [ConditionalTheory]
-        [QuarantinedTest]
+        [Theory]
```
Comment (ajaybhargavb) [inline src/ProjectTemplates/test/ByteOrderMarkTest.cs:24]: This is the actual code change here.

File: src/ProjectTemplates/test/ProjectTemplates.Tests.csproj [hunk 2]
```diff
@@ -31,6 +31,14 @@
     <Compile Include="..\Shared\**" LinkBase="Helpers" />
   </ItemGroup>
 
+  <ItemGroup>
+    <!-- We need the actual templates in the output directory for tests to verify file encodings. -->
+    <Content Include="..\Web.ProjectTemplates\**" LinkBase="Assets\Web.ProjectTemplates" />
+    <Content Include="..\Web.Spa.ProjectTemplates\**" LinkBase="Assets\Web.Spa.ProjectTemplates" />
+    <Content Include="..\Web.ItemTemplates\**" LinkBase="Assets\Web.ItemTemplates" />
+    <Content Include="..\Web.Client.ItemTemplates\**" LinkBase="Assets\Web.Client.ItemTemplates" />
+  </ItemGroup>
```
Comment (ajaybhargavb) [inline src/ProjectTemplates/test/ProjectTemplates.Tests.csproj:40]: Also this.

**Review by sharwell — DISMISSED** [review]
The **.editorconfig** needs to be updated as well, because it is currently configured to overwrite each of these changes the next time the file gets saved.

https://github.com/dotnet/aspnetcore/blob/be86316fcffdbf38039d2da7e512494aa17514a6/.editorconfig#L18

File: src/ProjectTemplates/test/ProjectTemplates.Tests.csproj (hunk 2 above)
Comment (pranavkm) [reply src/ProjectTemplates/test/ProjectTemplates.Tests.csproj:40]: Nice.

**Review by pranavkm — APPROVED** [review]

**Review by mkArtakMSFT — APPROVED** [review]

**Comment by ajaybhargavb** [comment]
@mkArtakMSFT this is ready to be merged

**Pull request merged**